File: app/src/main/java/com/example/cinemates/adapter/MovieViewHolder.java
package com.example.cinemates.adapter;

import androidx.annotation.NonNull;

import com.example.cinemates.databinding.ListItemMediaPosterBinding;
import com.example.cinemates.util.RecyclerViewEmptySupport;

/**
 * @author Antonio Di Nuzzo
 * Created 28/07/2022 at 08:51
 */
class MovieViewHolder extends RecyclerViewEmptySupport.ViewHolder {
    private static final String TAG = MovieViewHolder.class.getSimpleName();
    ListItemMediaPosterBinding mBinding;

    MovieViewHolder(@NonNull ListItemMediaPosterBinding listItemMediaPosterBinding) {
        super(listItemMediaPosterBinding.getRoot());
        this.mBinding = listItemMediaPosterBinding;

    }


}

File: app/src/main/java/com/example/cinemates/adapter/MovieViewHolder.kt
package com.example.cinemates.adapter

import androidx.recyclerview.widget.RecyclerView
import com.example.cinemates.databinding.ListItemMovieLongBinding
import com.example.cinemates.databinding.ListItemMovieSmallBinding

/**
 * @author Antonio Di Nuzzo
 * Created 28/07/2022 at 08:51
 */
class MovieViewHolder : RecyclerView.ViewHolder {
    var longBinding: ListItemMovieLongBinding? = null
    var smallBinding: ListItemMovieSmallBinding? = null

    constructor(listItemMovieLongBinding: ListItemMovieLongBinding) : super(listItemMovieLongBinding.root) {
        longBinding = listItemMovieLongBinding
    }

    constructor(listItemMovieSmallBinding: ListItemMovieSmallBinding) : super(
        listItemMovieSmallBinding.root
    ) {
        smallBinding = listItemMovieSmallBinding
    }
}